app/src/main/java/skills/future/planer/db/task/TaskData.java

```diff
@@ -1,66 +1,101 @@
 import skills.future.planer.db.task.enums.priority.Priorities;
 import skills.future.planer.db.task.enums.priority.TimePriority;
 
+/**
+ * Class encapsulate data for task
+ *
+ * @author Mikołaj Szymczyk
+ */
 @Getter
 @Setter
 @Entity(tableName = "taskData")
-public class TaskData implements Serializable {
+public class TaskData implements Parcelable {
+    /**
+     * task id
+     */
     @PrimaryKey(autoGenerate = true)
     private int taskDataId;
+    /**
+     * task status
+     */
     @ColumnInfo(name = "status")
     private Boolean status;
+    /**
+     * task category
+     */
     @ColumnInfo(name = "category")
     private TaskCategory category;
+    /**
+     * task priority
+     */
     @ColumnInfo(name = "priorities")
     private Priorities priorities;
+    /**
+     * task time priority
+     */
     @ColumnInfo(name = "timePriority")
     private TimePriority timePriority;
+    /**
+     * task title
+     */
     @ColumnInfo(name = "taskTitleText")
     private String taskTitleText;
+    /**
+     * task details
+     */
     @ColumnInfo(name = "taskDetailsText")
     private String taskDetailsText;
+    /**
+     * starting date of task period
+     */
     @ColumnInfo(name = "startingDate")
     private String startingDate = null;
+    /**
+     * ending date of task period
+     */
     @ColumnInfo(name = "endingDate")
     private String endingDate = null;
-
-    public TaskData(int taskDataId, Boolean status, int category, int priorities,
-                    int timePriority, String taskTitleText,String taskDetailsText,
-                    CalendarDay startingDate, CalendarDay endingDate) {
-        this.taskDataId = taskDataId;
-        this.status = status;
-        switch (category) {
-            case 0 -> this.category = TaskCategory.Private;
-            case 1 -> this.category = TaskCategory.Work;
-        }
-        switch (priorities) {
-            case 0 -> this.priorities = Priorities.Important;
-            case 1 -> this.priorities = Priorities.NotImportant;
+    /**
+     * Field used to pack taskData to bundle it isn't save in database
+     */
+    @Ignore
+    public static final Creator<TaskData> CREATOR = new Creator<>() {
+        @Override
+        public TaskData createFromParcel(Parcel in) {
+            return new TaskData(in);
         }
-        switch (timePriority) {
-            case 0 -> this.timePriority = TimePriority.Urgent;
-            case 1 -> this.timePriority = TimePriority.NotUrgent;
+
+        @Override
+        public TaskData[] newArray(int size) {
+            return new TaskData[size];
         }
-        this.taskTitleText = taskTitleText;
-        this.taskDetailsText = taskDetailsText;
-        setEndingCalendarDate(endingDate);
-        setStartingCalendarDate(startingDate);
-    }
+    };
 
+    /**
+     * Constructor of taskData
+     *
+     * @param category        could be private of word (take from enum TaskCategory)
```

Comment:
Popraw komentarz bo wprowadza zamieszanie

Reply:
to be private or not to be